File: 240707/최단 Run Length 인코딩/shortest-run-length-encoding.py
def RLE(arr):
    result = ""
    alp = arr[0]
    cnt = 1

    for i in range(1, len(arr)):
        if arr[i] != arr[i - 1]:
            result += alp
            result += str(cnt)
            alp = arr[i]
            cnt = 1
        else:
            cnt += 1
    result += alp
    result += str(cnt)

    return len(result)

arr = list(input().strip())
n = len(arr)
ans = 100

for i in range(n):
    rotated = [arr[(i + j) % n] for j in range(n)]
    ans = min(ans, RLE(rotated))

print(ans)

File: 240707/최단 Run Length 인코딩/shortest-run-length-encoding.js
function RLE(arr) {
    let ans = ""
    let alp = arr[0]
    let cnt = 1
    for (let i = 1; i < arr.length; i++){
        if (arr[i] != arr[i - 1]){
            ans += alp
            ans += cnt
            alp = arr[i]
            cnt = 1
        } else {
            cnt++
        }
    }
    ans += alp
    ans += cnt

    return ans
}

const fs = require("fs")
const input = fs.readFileSync(0).toString().trim().split("")
let len = input.length

for (let i = 0; i < input.length; i++){
    len = Math.min(len, RLE(input.slice(input.length - i).concat(input.slice(0, input.length - i))).length)
}

console.log(len)